frontend/src/components/form/SearchUsers.vue

```diff
@@ -0,0 +1,155 @@
+<script setup lang="ts">
+import { onMounted, ref, Ref } from 'vue';
+import { storeToRefs } from 'pinia';
+import Button from 'primevue/button';
+import Dropdown from 'primevue/dropdown';
+import RadioButton from 'primevue/radiobutton';
+
+import { useBucketStore, useConfigStore, useUserStore } from '@/store';
+import { Regex } from '@/utils/constants';
+import type { IdentityProvider, User } from '@/interfaces';
+
+// Store
+const userStore = useUserStore();
+const { permissions } = storeToRefs(useBucketStore());
+const { config } = storeToRefs(useConfigStore());
+const { userSearch } = storeToRefs(useUserStore());
+
+// State
+const selectedIDP: Ref<IdentityProvider | null> = ref(null);
+const selectedUser: Ref<User | null> = ref(null);
+const selectedUserIsInvalid: Ref<boolean> = ref(false);
+const userSearchInput: Ref<string | undefined> = ref('');
+
+// Emits
+const emit = defineEmits(['add-user', 'cancel-search-users']);
+
+// Functions
+const getUserDropdownLabel = (option: User) => {
+  if( selectedIDP.value?.idp ) {
+    if( selectedIDP.value.searchable ) {
+      return `${option.fullName} [${option.email}]`;
+    }
+    else {
+      return option.email;
+    }
+  }
+};
+
+const onAdd = () => {
+  emit('add-user', selectedUser.value);
+  onReset();
+};
+
+const onCancel = () => {
+  onReset();
+  emit('cancel-search-users');
+};
+
+const onChange = (event: any) => {
+  // Duplicate user check
```

Comment:
In CHEFS I don't think we ever did a duplicate user check - the autocomplete would still show the user even though they already existed, but instead if you tried to re-add them it was effectively a no-op because the entry is already there. If this is pre-cleaning out the dropdown list in advance then that's a reasonable 👍 in my books.

Reply:
i believe this is leaving existing users in dropdown but filed turn red if you select one. i'm ok with this too

Reply:
Currently existing users will appear, but you are unable to add them. The dropdown will turn red and the Add button is disabled. I believe future UX considerations will update this error handling.